src/Sylius/Component/Taxonomy/Model/Factory/TaxonomyFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Taxonomy\Factory;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Model\TaxonomyInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class TaxonomyFactory implements TaxonomyFactoryInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    protected $repository;
+
+    /**
+     * @param RepositoryInterface $repository
+     */
+    public function __construct(RepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @param TaxonInterface $taxon
+     * @return TaxonomyInterface
+     */
+    public function create(TaxonInterface $taxon)
+    {
+        $taxonomy = $this->repository->createNew();
+        $taxonomy->setRoot($taxon);
+
+        return $taxonomy;
+    }
+}
```

Comment:
I'm not sure about `Model\Factory` namespace, it is a factory but not a model. I think `Factory` namespace in would be more intuitive.